### DISCUSSION TITLE
ReferenceError: theme is not defined when used in tailwind.config

### DISCUSSION THREAD

**Discussion by J4v4Scr1pt** [tailwindlabs/tailwindcss#12765] [Help] [answered]
Im trying to create a custom component class inside of the tailwind.config.
But after using theme function I get the error below. As I understand it after som googling you should config the postcss file to make it work. But after reading the [documentation](https://tailwindcss.com/docs/using-with-preprocessors#variables) it's seems like this is not needed?!

Could someone please help me understand how I can use theme() inside of the tailwindconfig.

postcss.config:
```
module.exports = {
	plugins: {
		tailwindcss: {},
		autoprefixer: {},
	},
};
```

The component I try to add:
```
		addComponents({
				'.btn-primary': {
					color: theme(colors.purple.500),
					borderRadius: '.25rem',
					fontWeight: '600',
				},
			});
```

```
./app/globals.css.webpack[javascript/auto]!=!./node_modules/.pnpm/registry.npmjs.org+next@14.0.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/build/webpack/loaders/css-loader/src/index.js??ruleSet[1].rules[11].oneOf[12].use[2]!./node_modules/.pnpm/registry.npmjs.org+next@14.0.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/build/webpack/loaders/postcss-loader/src/index.js??ruleSet[1].rules[11].oneOf[12].use[3]!./app/globals.css
ReferenceError: theme is not defined
```

**Comment by wongjn** [comment, marked as answer]
As the error states, consider checking you have `theme()` defined as a function in the scope that you call `addComponents()`. For example, when passing your function to the `plugin()` function, ensure you have the destructed `theme` member:

```js
const plugin = require('tailwindcss/plugin');

// …

//                       This!
//                         ↓
plugin(({ addComponents, theme }) => {
```

**Reply by J4v4Scr1pt** [reply]
Thx man!
Damn I went back to the docs about adding custom components.. and there it was.. 
![image](https://github.com/tailwindlabs/tailwindcss/assets/5929055/c0e7b473-7455-4444-8d85-2abec1579e97)

How stupid of me 😅

Have a superb day!
